src/client/pmi1.c

```diff
@@ -59,20 +61,37 @@ static int pmi_init = 0;
 
 int PMI_Init(int *spawned)
 {
+    pmix_status_t rc = PMIX_SUCCESS;
     pmix_value_t *val;
-    pmix_status_t rc;
+    pmix_proc_t proc;
+    pmix_info_t info[2];
+    bool  val_optinal = 1;
+    int val_timeout = PMIX_TIMEOUT_VALUE;
 
     if (PMIX_SUCCESS != PMIx_Init(&myproc)) {
         return PMI_ERR_INIT;
     }
 
+    /* getting internal key requires special rank value */
+    memcpy(&proc, &myproc, sizeof(myproc));
+    proc.rank = PMIX_RANK_UNDEF;
+
+    /* set controlling parameters
+     * PMIX_OPTIONAL - expect that these keys should be available on startup
+     * PMIX_TIMEOUT - meet legacy PMI requirement
+     */
+    PMIX_INFO_CONSTRUCT(&info[0]);
+    PMIX_INFO_CONSTRUCT(&info[1]);
+    PMIX_INFO_LOAD(&info[0], PMIX_OPTIONAL, &val_optinal, PMIX_BOOL);
+    PMIX_INFO_LOAD(&info[1], PMIX_TIMEOUT, &val_timeout, PMIX_INT);
```

Comment:
One other point, just to be sure there is understanding. The _optional_ flag tells the client to not  request the data from the server if the client doesn't already have it. So giving both the _optional_flag plus a timeout or request to check the server won't accomplish anything - once we see the _optional_ flag, we'll never ask the server. This is why I am suggesting a new key that indicates we should check both client and local server, but don't wait or try to find the data elsewhere if it isn't immediately present.